src/structures/gdt.rs

```diff
@@ -105,14 +105,24 @@ impl GlobalDescriptorTable {
         }
     }
 
-    /// Forms a GDT from a raw table and a next_free counter.
+    /// Forms a GDT from a slice of `u64`.
     ///
     /// # Safety
     ///
-    /// The user must ensure that the provided table contains well formed entries
-    /// and that the value of `next_free` does not exceed the maximum size of the table.
+    /// * The user must make sure that the entries are well formed
+    /// * The provided slice **must not be larger than 8 items** (only up to the first 8 will be observed.)
     #[inline]
-    pub const unsafe fn from_raw_parts(table: [u64; 8], next_free: usize) -> GlobalDescriptorTable {
+    pub const unsafe fn from_raw_parts(slice: &[u64]) -> GlobalDescriptorTable {
+        let next_free = if slice.len() >= 8 { 8 } else { slice.len() };
```

Comment:
I would prefer to panic on slices longer than 8, even if this means that we can only provide this method on nightly. Silently ignoring additional entries seems a bit dangerous.